### PR TITLE
Learn search results list dates are different than default drawer details dates

### PR DESCRIPTION
**Pull request by JenniWhitman**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
closes [3842](https://github.com/mitodl/open-discussions/issues/3842)

#### What's this PR do?
Swaps both the ol-search-ui and open-discussions front end repos off using best_start_date and onto start_date as best_start_date is often enrollment date and a mismatch... and misnomer.

#### How should this be manually tested?
search for courses in open-discussions/learn/search and compare against the database to find preferably courses with mismatched best_start_date and start_date - I would recommend:

- 1 that is currently running with past dates
- 1 that is currently running with future dates
- 1 that is currently running only
- 1 that ran in the past without future dates
- 1 that ran in the past but has future dates
- 1 that runs in the future only

The selected "best run" should be the one expected which would be:

- Currently running (today's date is between the start and end dates)
- the next run (start date is after today's date, there is no current run)
- the last run that ran if neither of the previous exist.

The date on the search results should match the date in the drawer that opens when you click on the course
The dates in the date drop down (if there are multiple runs) should be in order. This can be hard to tell as sometimes there are wacky end dates. To fix this, I modified line 207 of expandedLearningResourceDisplay.js to `{getStartDate(object, run)} - {run.end_date}` so I could see this while testing.

#### Where should the reviewer start?
Start with the js changes as those are what is on learn/search and what is referenced in the ticket.

#### Any background context you want to provide?
You know that dog flying hte plane meme? That's me. I'm not certain that best_start_date doesn't have a purpose, only that that date causes the mismatch and because of inconsistencies with saved data vs what we display elsewhere. I also made the changes to both the js and ts versions since this would be incorrect in search-ui going forward if change isn't made there as well, although it should work the same in both.

### CODE REVIEW

**Comment by ChristopherChudzicki** [comment]
@JenniWhitman Re `best_start_date` I have found this confusing in the past. Matt clarified for me:

> best_start_date is specific to each run, and was meant to handle cases from other sources where there were multiple start dates, some of which might be blank, for example:
> 
> ```
> course_run.best_start_date = (
>     course_run.enrollment_start
>     or course_run.start_date
>     or semester_year_to_date(course_run.semester, course_run.year)
> )
> ```
> 
> But for prolearn there’s only 1 start date, so that’s what best_start_date should be too.



https://github.com/mitodl/open-discussions/pull/3804#discussion_r1068540384

**Comment by JenniWhitman** [comment]
> @JenniWhitman Re `best_start_date` I have found this confusing in the past. Matt clarified for me:
> 
> > best_start_date is specific to each run, and was meant to handle cases from other sources where there were multiple start dates, some of which might be blank, for example:
> > ```
> > course_run.best_start_date = (
> >     course_run.enrollment_start
> >     or course_run.start_date
> >     or semester_year_to_date(course_run.semester, course_run.year)
> > )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But for prolearn there’s only 1 start date, so that’s what best_start_date should be too.
> 
> [#3804 (comment)](https://github.com/mitodl/open-discussions/pull/3804#discussion_r1068540384)

The problem is, the code actually prefers enrollment start date to best start date - I COULD make it so it grabs another if it's not there, but quite often I'm seeing it use enrollment date even if there is a start date which is the reason for the discrepency.

File: frontends/ol-search-ui/src/util.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const compareRuns = (
 const findBestRun = (
   runs: LearningResourceRun[]
 ): LearningResourceRun | undefined => {
-  const dated = runs.filter(run => run.best_start_date && run.best_end_date)
+  const dated = runs.filter(run => run.start_date && run.end_date)
```
Comment (mbertrand) [inline frontends/ol-search-ui/src/util.tsx:78]: I think doing this instead would make it more consistent with `ExpandedLearningResourceDisplay.tsx`:

```javascript
  const startDate =
    hasCertificate && bestAvailableRun ?
      getStartDate(resource.platform ?? "", bestAvailableRun).format(DISPLAY_DATE_FORMAT) :
      null
```

File: frontends/open-discussions/src/components/LearningResourceCard.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ export function LearningResourceDisplay(props: Props) {
   const hasCertificate = object.certification && object.certification.length > 0
   const startDate =
     hasCertificate && bestAvailableRun
-      ? moment(bestAvailableRun.best_start_date).format(DISPLAY_DATE_FORMAT)
+      ? moment(bestAvailableRun.start_date).format(DISPLAY_DATE_FORMAT)
```
Comment (mbertrand) [inline frontends/open-discussions/src/components/LearningResourceCard.js:151]: Similar to above:

```javascript
  const startDate =
    hasCertificate && bestAvailableRun
      ? getStartDate(object, bestAvailableRun)
      : null
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: frontends/ol-search-ui/src/util.tsx (hunk 1 above)
Comment (JenniWhitman) [reply frontends/ol-search-ui/src/util.tsx:78]: you're not wrong - that better solves the "and what if there's none" problem

**Review by mbertrand — APPROVED** [review]
Looks good! 👍

**Pull request merged**